packages/record/src/components/CopyToClipboardButton/index.js

```diff
@@ -0,0 +1,152 @@
+import React, { useRef, useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { ButtonIcon } from 'react-rainbow-components';
+import InternalTooltip from 'react-rainbow-components/components/InternalTooltip';
+import { useScrollingIntent } from '@rainbow-modules/hooks';
+import { CopyToClipboard } from '@rainbow-modules/icons';
+import { isSupported, copy } from './helpers';
+
+/** A button with modern approach to copy text to clipboard. Would try to use execCommand with fallback to IE specific clipboardData interface. */
+export default function CopyToClipboardButton(props) {
+    const {
+        style,
+        id,
+        className,
+        value,
+        copyText,
+        copiedText,
+        variant,
+        size,
+        shaded,
+        disabled,
+        tabIndex,
+        assistiveText,
+    } = props;
+    const triggerRef = useRef();
+    const tooltipRef = useRef();
+    const [isVisible, setVisible] = useState(false);
+    const [isCopied, setCopied] = useState(false);
+    const text = isCopied ? copiedText : copyText;
+
+    useEffect(() => {
+        if (isCopied) {
+            setTimeout(() => {
+                setVisible(false);
+                setCopied(false);
+            }, 3000);
+        }
+    }, [isCopied]);
+
+    useScrollingIntent({
+        callback: () => setVisible(false),
+        isListening: isVisible,
+        triggerElementRef: () => triggerRef.current.htmlElementRef,
+    });
+
+    if (!isSupported()) {
+        return null;
+    }
+
+    const handleClick = () => {
+        const success = copy(value);
+        triggerRef.current.focus();
```

Comment:
why need to focus manually?

Reply:
In the copy function when doing this:

``` js
const range = document.createRange ();
const selection = document.getSelection ();

range.selectNodeContents (mark);
selection.addRange (range);
```

the text is selected to copy it, the focus moves to this element so it must be changed manually.